modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4MessageChannelHandler.java

```diff
@@ -144,7 +149,13 @@ private void doFlush(ChannelHandlerContext ctx) throws IOException {
                 break;
             }
             final WriteOperation write = currentWrite;
-            final ByteBuf currentBuffer = write.buffer();
+            final ByteBuf currentBuffer;
+            try {
+                currentBuffer = write.buffer();
+            } catch (Exception e) {
+                logger.error(new ParameterizedMessage("Failed to serialize message to [{}]", ctx.channel().remoteAddress()), e);
```

Comment:
I'm pretty shaky on this area of the code and don't have the capacity to dig deeply right now, but if we swallow the exception here does that mean that the response handler isn't notified? If so, that seems bad.

Reply:
Yup, you are 100% correct here, we still need to send a failure response.
Also for request sending we need to bubble up the failure which as it turns out requires quite the refactoring since sending the response isn't a sync thing.
I have a prototype of how this could work in here now (and a test reproducing the issue for both sending requests and responses) but it'll take me a bit of time to clean it up properly.
On the bright side this kind of refactoring opens up some new possibilities for monitoring the networking.